### ISSUE TITLE
'no-unneeded-ternary' does not catch ternary issues with data properties

### ISSUE THREAD

**Issue by BrandonYeager** [eslint/eslint#11852]
**Tell us about your environment**

* **ESLint Version: 5.16.0**
* **Node Version: 12.4.0**
* **npm Version: 6.9.0**

**What parser (default, Babel-ESLint, etc.) are you using? 'babel-eslint'**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  root: true,
  env: {
    node: true,
  },
  'extends': [
    'plugin:vue/recommended', 'eslint:recommended',
  ],
  rules: {
    'no-unneeded-ternary': ['error', { defaultAssignment: false }], // Fixable
  },
  parserOptions: {
    parser: 'babel-eslint',
  },
};

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
let testObject = {
    fieldValue: 1,
}

testObject.fieldValue ? testObject.fieldValue : 0
```

```bash
vue-cli-service lint --no-fix
```

**What did you expect to happen?**
```
Unnecessary use of conditional expression for default assignment.eslint (no-unneeded-ternary)
```

**What actually happened? Please include the actual, raw output from ESLint.**
```
No lint errors found!
```

**Are you willing to submit a pull request to fix this bug?**
```
No (Not enough experience with JS)
```

**Comment by platinumazure** [comment]
Hi @BrandonYeager, thanks for the issue.

This is working as designed, because ESLint cannot detect via static analysis whether `testObject.fieldValue` is a getter with side effects. If that were the case, then invoking the property twice (in a ternary) vs once (in a logical or expression) would change the behavior of the code.

I could support a new rule option which would configure the rule to warn on property accesses anyway, so that users could at least opt in to this functionality if they know they aren't using getters or are willing to accept the risk of false positives that would need disable comments.

**Comment by BrandonYeager** [comment]
Thanks for the explanation and the quick reply!

I think a new rule option to make it work with properties would be great. We're currently using vue and almost all of our variables are data properties.
